ArduCopter/mode_loiter_poi.cpp

```diff
@@ -0,0 +1,217 @@
+#include "Copter.h"
+
+#if MODE_LOITER_ENABLED == ENABLED
+
+/*
+ * Init and run calls for loiter flight mode
+ */
+
+// loiter_init - initialise loiter controller
+bool ModeLoiter_POI::init(bool ignore_checks)
+{
+    bool isExist = false;
+
+    // Find MAV_CMD_DO_SET_ROI and set ROI position by command
+    uint16_t mission_count = copter.mode_auto.mission.num_commands();
```

Comment:
Is this flight mode part of the Auto mode?
If it is part of Auto-mode, it will be added to the Auto-mode class.

Reply:
No, I want to add mission on Loiter_Poi class, but  instance of the mission is restricted so that there can only be one in the system.
We can remove this line if get any other way to set POI position. (ex. using RC channel)